Configuration/StandardSequences/python/GeneratorEmbMix_cff.py

```diff
@@ -0,0 +1,14 @@
+import FWCore.ParameterSet.Config as cms
+
+# Produce GenParticles of the two HepMCProducts
+from Configuration.StandardSequences.Generator_cff import *
+from GeneratorInterface.Core.generatorSmeared_cfi import *
+genParticles.mix = cms.string("mix")
+genParticles.doSubEvent = cms.untracked.bool(True)
+genParticles.useCrossingFrame = cms.untracked.bool(True)
+genParticles.saveBarCodes = cms.untracked.bool(True)
+genParticles.abortOnUnknownPDGCode = cms.untracked.bool(False)
```

Comment:
instead of changing the parameters of an existing modules, you should clone the module with the changed parameters (see https://github.com/cms-sw/cmssw/pull/30136#pullrequestreview-426297630 )

Reply:
Dear @silviodonato , if I understand correct, You mean something like:
`genParticlesEmb = genParticles.clone(
        mix = cms.string("mix"),
        doSubEvent = cms.untracked.bool(True),
        useCrossingFrame = cms.untracked.bool(True),
        saveBarCodes = cms.untracked.bool(True),
        abortOnUnknownPDGCode = cms.untracked.bool(False)
)

GenSmeared = cms.Sequence(generatorSmeared)
hiGenJets = cms.Sequence(genParticlesForJets*hiRecoGenJets)
pgen = cms.Sequence(cms.SequencePlaceholder("mix")+cms.SequencePlaceholder("generator")+cms.SequencePlaceholder("randomEngineStateProducer")+VertexSmearing+GenSmeared+genParticlesEmb+hiGenJets)
` 
Isn't it?
If so, it fails with:
'''
----- Begin Fatal Exception 19-Jun-2020 11:21:23 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'generation_step'
   [2] Calling method for module InputGenJetsParticleSelector/'genParticlesForJets'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for a container with elements of type: reco::Candidate
Looking for module label: genParticles
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------

'''

Reply:
I've just realized that this file is almost a copy of `Configuration/StandardSequences/python/GeneratorMix_cff.py`.
Then, you can use load directly `GeneratorMix_cff` with
```
import FWCore.ParameterSet.Config as cms
from Configuration.StandardSequences.GeneratorMix_cff import *

pgen = cms.Sequence(cms.SequencePlaceholder("mix")+cms.SequencePlaceholder("generator")+cms.SequencePlaceholder("randomEngineStateProducer")+VertexSmearing+GenSmeared+genParticles+hiGenJets)

```

Reply:
Thank You! Done.